internal/provider/filemanager_directory_resource.go

```diff
@@ -111,14 +110,23 @@ func (r *filemanagerDirectoryResource) Read(ctx context.Context, req resource.Re
 
 	metadata, err := r.client.GetFileMetadata(ctx, directoryPath)
 	if err != nil {
-		if strings.HasSuffix(err.Error(), "404") {
-			// If the directory is not found, we have drift, and it was deleted from qbee
-			resp.State.RemoveResource(ctx)
-		} else {
+		errorHandled := false
+		if clientErr, ok := err.(client.Error); ok {
+			if errObj, ok := clientErr["error"].(map[string]any); ok {
+				if code, ok := errObj["code"].(float64); ok && int(code) == 404 {
+					// If the directory is not found, we have drift, and it was deleted from qbee
+					tflog.Info(ctx, fmt.Sprintf("Directory %v not found, removing from state", directoryPath))
+					resp.State.RemoveResource(ctx)
+					errorHandled = true
```

Comment:
@lesteenman have you considered to return after this statement to avoid having `errorHandled` variable?

Reply:
That actually works great in the existing flow, making the change.